src/app/clusters/network-commissioning/network-commissioning.cpp

```diff
@@ -31,6 +31,7 @@
 #include <platform/PlatformManager.h>
 #include <platform/internal/DeviceNetworkInfo.h>
 #include <trace/trace.h>
+#include <vector>
```

Comment:
This change has a quite big flash impact, supposedly because of using STL. Perhaps, you could use `ScopedBuffer` and simple insertion sort with ignoring duplicated results to lower these numbers. Btw. which platform do you test it on? Does OpenThread return duplicated results?

Reply:
Thank you for your review! 🙂 

> This change has a quite big flash impact, supposedly because of using STL.

I checked whether STL was already being used and used it because I added STL to this file commonly used by commissionee manufacturers.

https://github.com/project-chip/connectedhomeip/blob/15ac91713ad51bee8fed30f956388eeec3f5f55a/src/app/ClusterStateCache.h#L30-L34

As far as I know, above common code is already using STL. Please let me know if I'm misunderstanding something.

> Perhaps, you could use `ScopedBuffer` and simple insertion sort with ignoring duplicated results to lower these numbers.

Could you please check first if I can use STL in this file? In my opinion, if I could use STL it would be simpler to implement than `ScopedBuffer`. I agree with your comment that it is better to use insertion sort. 👍 I'll implement insertion sort after deciding which data structure to use.

> Btw. which platform do you test it on? Does OpenThread return duplicated results?

I tested with the lock-app example on efr32 platform.

- **Board**: Wireless Starter Kit Mainboard (BRD4001A Rev A01)
- **Board**: EFR32MG12 2.4 GHz 10 dBm Radio Board (BRD4162A Rev A02)
- **Target Part**: EFR32MG12P332F1024GL125

I checked the scan result with the log output by the code below on the device. I'm not familiar with the OpenThread code, but it seems that OpenThread returns duplicated results. Please let me know if you need more information.

https://github.com/project-chip/connectedhomeip/blob/15ac91713ad51bee8fed30f956388eeec3f5f55a/src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp#L473-L474

When I checked the log, the number of duplicate results was slightly different each time it performed scan. Last time I checked, there were about 2 duplicate results out of 50. If `panId` and `extendedPanId` of the two results were the same, I treated them as duplicates, and RSSIs of them were different.

Reply:
STL is generally used in controller applications which are rarely implemented on embedded platforms. Embedded platforms tend not to use STL containers on the other hand. You may verify with a tool such as https://pypi.org/project/elf-diff/ which functions were added to the final binary by your change, but my guess is that most of that is `std::vector` given that the increase on platforms such as EFR32 is over 2kB. I think it should be possible to cut this number significantly :)

Reply:
Good. Following the convention of embedded platforms, I'll implement using `ScopedBuffer` instead of STL and compare binary sizes next week.

Reply:
+1 - STL is not acceptable in embedded code except with very small exceptions. We also try to avoid heap if possible ... but that is not always possible especially for somewhat larger one-use operations.

Reply:
Thanks for the advice. I'll be careful when implementing the embedded code from now on.

Reply:
There is also a bubble sort provided here in the sdk: https://github.com/project-chip/connectedhomeip/blob/master/src/lib/support/SortUtils.h

This was introduced specifically for its low flash cost.  That is probably a good option here.

Reply:
@Damian-Nordic , @andy31415 I changed it to use `ScopedBuffer` instead of STL and use insertion sort. Could you please review again? The reason the code is a bit complicated is that if there are duplicate results, the result with higher RSSI is used.

@msandstedt Thanks for the suggestion! I used insertion sort because I thought insertion sort performed better when inserting new item into sorted list.

Reply:
Use of STL containers was removed and the sort was reworked.  Hence, resolved.